### PR TITLE
fix(cli): include JSON modules in bundle

### LINKED ISSUE #13150: `deno bundle` not working with JSON module
I'm testing out the new JSON modules functionality introduced with `1.17.0`.  I'm expecting the `deno bundle` command to be able to handle the JSON module import, but I'm getting an error.  Is this a bug, or is bundling JSON modules not supported?

**./test.ts**
```typescript
import j from './test.json' assert { type: 'json' }
console.log(JSON.stringify(j))
```

**./test.json**

```json
{"foo":"bar"}
```

Console session:

```shell
$ deno --version
deno 1.17.0 (release, x86_64-pc-windows-msvc)
v8 9.7.106.15
typescript 4.5.2

$ deno run test.ts
{"foo":"bar"}

$ deno bundle test.ts
Check file:///C:/.../test.ts
Bundle file:///C:/.../test.ts
error: Unable to output during bundling.

Caused by:
    0: load_transformed failed
    1: Bundler.load() failed
    2: Bundler.loader.load(file:///C:/.../test.json) failed    
    3: Expected ';', '}' or <eof> at file:///C:/.../test.json:1:7
```

### PR DESCRIPTION
**Pull request by kitsonk**
Fixes #13150

### CODE REVIEW

**Comment by andreubotella** [comment]
Note that this will output incorrect sourcemaps with `Deno.emit()`, since the byte offsets are into the mapped source. I tried to hack around this with a custom implementation of [`Input`](https://docs.rs/swc_common/latest/swc_common/input/trait.Input.html) that treated all positions before the start of the JSON as 0, but I didn't get very far with it.

This is not a blocker for this PR, but it should probably be fixed before the next patch release.

File: cli/ast/mod.rs [hunk 1]
```diff
@@ -350,12 +350,23 @@ pub fn transpile_module(
   options: &EmitOptions,
   cm: Rc<SourceMap>,
 ) -> Result<(Rc<deno_ast::swc::common::SourceFile>, Module), AnyError> {
-  let source = strip_bom(source);
```
Comment (bartlomieju) [inline cli/ast/mod.rs:353]: I believe BOM should be stripped for JSON too, we do that in `deno_core`:https://github.com/denoland/deno/blob/de2df20d574b3a311d80d628cb7cefe20020d5e7/core/modules.rs#L783

**Review by bartlomieju — APPROVED** [review]
LGTM, Andreu has c good point regarding source maps.

File: cli/ast/mod.rs [hunk 2]
```diff
@@ -350,12 +350,23 @@ pub fn transpile_module(
...
+  let source = if media_type == MediaType::Json {
+    format!(
+      "export default JSON.parse(`{}`);",
+      source.replace("${", "\\${")
```
Comment (dsherret) [inline cli/ast/mod.rs:356]: Also should escape backtick?

It's funny, in dnt when implementing this feature I forgot the opposite (now fixed).

**Review by dsherret — APPROVED** [review]
LGTM once the two issues are fixed.

**Comment by kitsonk** [comment]
> Note that this will output incorrect sourcemaps with `Deno.emit()`, since the byte offsets are into the mapped source. I tried to hack around this with a custom implementation of [`Input`](https://docs.rs/swc_common/latest/swc_common/input/trait.Input.html) that treated all positions before the start of the JSON as 0, but I didn't get very far with it.
> 
> This is not a blocker for this PR, but it should probably be fixed before the next patch release.

Yeah, I realised it when I did it that it would "mess" things up, but I plan to mop it up, along with other things as part of #13021.

File: cli/ast/mod.rs (hunk 1 above)
Comment (kitsonk) [reply cli/ast/mod.rs:353]: Yeah, I was thinking it was the "strip hashbang" logic for some reason, not the strip BOM stuff... 🤦

**Pull request merged**